### PR TITLE
refactor!: replace eris with discord.js

### PR DESCRIPTION
**Pull request by mikolajkalwa**

### CODE REVIEW

**Review by wiktorkowalski — COMMENTED** [review]
change PR title from `refactor!: raplace eris with discord.js` to `refactor!: replace eris with discord.js`

File: package.json [hunk 1]
```diff
@@ -1,37 +1,44 @@
 {
   "name": "games-deals-bot",
-  "version": "1.0.0",
+  "version": "3.0.0",
```
Comment (wiktorkowalski) [inline package.json:3]: much breaking changes

File: src/bot/commands/webhook/webhook.definition.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      .addBooleanOption((option) => option
+        .setName('blacklist')
+        .setDescription('Select true to clear blacklist'))))
+  .addSubcommand((deleteSubcommand) => deleteSubcommand
+    .setName('delete')
+    .setDescription('Delete webhook')
+    .addStringOption((option) => option
+      .setName('webhook')
+      .setDescription('ID of webhook to edit')
+      .setRequired(true)));
```
Comment (wiktorkowalski) [inline src/bot/commands/webhook/webhook.definition.ts:70]: image

File: src/gd-api-client/index.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import GamesDealsApiClient from './gd-api.client';
+
+const gpApi = new GamesDealsApiClient();
```
Comment (wiktorkowalski) [inline src/gd-api-client/index.ts:3]: what does `gp` stands for? did you mean `gd`?

File: src/scripts/setupInteractions.ts [hunk 4]
```diff
@@ -1,17 +1,27 @@
...
-  console.log('Ready!');
-  const definitions = Array.from(commands.values()).map((x) => x.definition);
-  await bot.bulkEditGuildCommands('527065903926345728', definitions);
-  const existingCommands = await bot.getCommands();
-  console.log(existingCommands);
-  process.exit(0);
-});
+const definitions = Array.from(commands.values()).map((x) => x.definition.toJSON());
+const clientId = '904077341846818888';
+const guildId = '527065903926345728';
```
Comment (wiktorkowalski) [inline src/scripts/setupInteractions.ts:9]: should this be hard coded?

File: src/bot/commands/webhook/create.command.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      blacklist: blacklist || undefined,
+    });
+
+    await interaction.reply(`Webhook created succesfully\n${printWebhookDetails(savedWebhook)}`);
+  } catch (err) {
+    if (err instanceof DiscordAPIError) {
+      if (err.code === Constants.APIErrors.MISSING_PERMISSIONS) {
+        return await interaction.reply({ content: 'Please verify whether bot can manage webhooks in selected channel.', ephemeral: true });
+      }
+    }
```
Comment (wiktorkowalski) [inline src/bot/commands/webhook/create.command.ts:44]: what if there's other error?

**Review by wiktorkowalski — CHANGES_REQUESTED** [review]

File: src/bot/commands/webhook/create.command.ts (hunk 5 above)
Comment (mikolajkalwa) [reply src/bot/commands/webhook/create.command.ts:44]: it's thrown one line below and catched in here: https://github.com/mikolajkalwa/GamesDealsBot/blob/discord-js-switch/src/bot/bot.ts#L61

File: src/scripts/setupInteractions.ts (hunk 4 above)
Comment (mikolajkalwa) [reply src/scripts/setupInteractions.ts:9]: it should not

File: src/bot/commands/webhook/create.command.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      channel: webhook.channelId,
+      keywords: keywords || undefined,
+      blacklist: blacklist || undefined,
+    });
+
+    await interaction.reply(`Webhook created succesfully\n${printWebhookDetails(savedWebhook)}`);
+  } catch (err) {
+    if (err instanceof DiscordAPIError) {
+      if (err.code === Constants.APIErrors.MISSING_PERMISSIONS) {
+        return await interaction.reply({ content: 'Please verify whether bot can manage webhooks in selected channel.', ephemeral: true });
```
Comment (pjamroziak) [inline src/bot/commands/webhook/create.command.ts:42]: Why do you return only when the exception is raised?

File: src/bot/commands/webhook/edit.command.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    case 'set': {
+      await setPropertiesSubCommand(interaction, targetWebhook);
+      break;
+    }
+    default: {
+      await interaction.reply({ content: 'Unknown subcommand', ephemeral: true });
+      logger.warn(`Unknown subcommand: ${subcommand}`);
+    }
+  }
```
Comment (pjamroziak) [inline src/bot/commands/webhook/edit.command.ts:63]: I think the strategy pattern with dictionary is much cleaner

File: src/bot/commands/webhook/webhook.handler.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      case 'set':
+      case 'clear': {
+        await editWebhook(interaction, logger);
+        break;
+      }
+      default: {
+        logger.warn(`Unknown subcommand: ${subcommand}`);
+        await interaction.reply({ content: 'Unknown subcommand', ephemeral: true });
+      }
+    }
```
Comment (pjamroziak) [inline src/bot/commands/webhook/webhook.handler.ts:41]: Again, strategy pattern

**Review by pjamroziak — APPROVED** [review]

File: src/bot/commands/webhook/create.command.ts (hunk 6 above)
Comment (mikolajkalwa) [reply src/bot/commands/webhook/create.command.ts:42]: so the throw statement below is not executed in this case

**Pull request merged**